common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,15 +105,25 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+
         if n > 3:
             position_list = []
+            masquerading_student = False
```

Comment:
Please use a variable for `masquerade_settings = request.session.get('masquerade_settings')`

Reply:
we will only calculate this if request object is coming in.